runtime/lua/vim/lsp.lua

```diff
@@ -1407,6 +1455,26 @@ function lsp.buf_request_all(bufnr, method, params, callback)
   return cancel
 end
 
+--- Utility function for common use case of a buffer request at the current position.
+--- The position parameters are computed for each client depending on its `offset_encoding`.
+---@param method (string) see |vim.lsp.buf_request()|
+---@param _params (optional, table) optional table to be merged into position parameters
+---@param handler (function) see |vim.lsp.buf_request()|
+---@returns (function) see |vim.lsp.buf_request()|
+function lsp.buf_request_position(method, _params, handler)
```

Comment:
I'm not really a fan that we're special casing one params variant.

Not sure what to do about it - but I feel like this isn't going in the right direction.

Reply:
This should be local to `buf.lua`, but I put it here because otherwise we'd have to copy code into `lsp.omnifunc`. IMO that should also be in `buf.lua`.

Reply:
I don't think omnifunc should be in buf.lua, I think like the other vim compatible functions it should be in its own file being exposed via lsp.lua (see my recent tagfunc PR)